backend/tests/legacy/test_api_snapshot.py

```diff
@@ -103,3 +105,150 @@ def test_allocations(
         for key, val in allocation.items():
             if key == "amount":
                 assert int(val) == 1000
+
+
+@pytest.mark.api
+def test_deposit_basics(
+    client: Client, deployer: UserAccount, ua_alice: UserAccount, ua_bob: UserAccount
+):
+    # Check effective deposit before first GLM lock
+    (
+        total_effective_estimated_before_lock,
+        response_code,
+    ) = client.get_total_effective_estimated()
+    assert int(total_effective_estimated_before_lock["totalEffective"]) == 0
+    assert response_code == 200
+
+    user_deposit1_before_lock, response_code = client.get_user_deposit(
+        ua_alice.address, 1
+    )
+    assert int(user_deposit1_before_lock["effectiveDeposit"]) == 0
+    assert response_code == 200
+
+    (
+        user_estimated_deposit_before_lock,
+        response_code,
+    ) = client.get_user_estimated_effective_deposit(ua_alice.address)
+    assert int(user_estimated_deposit_before_lock["effectiveDeposit"]) == 0
+    assert response_code == 200
+
+    # Lock GML for one account
+    alice_GLM_budget = 10000
+    alice_first_lock = 8000
+    deployer.transfer(ua_alice, alice_GLM_budget)
+    ua_alice.lock(alice_first_lock)
+
+    time.sleep(5)
+
```

Comment:
I think we can do way better than this. 

1. Rename "wait_for_sync", so it is clear that it waits for a particular epoch.
2. Create wait_for_sync which will compare "blockchainHeight" with "indexedHeight" using the same API endpoint.